Aplib.Core/Collections/ExposedQueue.cs

```diff
@@ -65,7 +65,16 @@ public ExposedQueue(T[] array, int? count = null)
             MaxCount = array.Length;
             _array = array;
             _head = MaxCount - 1;
-            Count = count ?? MaxCount;
+            Count = count;
+        }
+
+        /// <inheritdoc cref="ExposedQueue{T}(T[],int)"/>
+        public ExposedQueue(T[] array)
+        {
+            MaxCount = array.Length;
+            _array = array;
+            _head = MaxCount - 1;
+            Count = MaxCount;
         }
```

Comment:
To stay up-to date with the other constructor, it would be better to call `this` from this constructor.

```suggestion
        /// <inheritdoc cref="ExposedQueue{T}(T[],int)"/>
        public ExposedQueue(T[] array)
            : this(array, array.Length)
        {
        }
```

Reply:
Good call, I was looking at this yesterday but I guess my brain was too fried.